docs/rfcs/0003-etag-changes.md

```diff
@@ -0,0 +1,226 @@
+
+### Caching
+
+Currently, the Etag is saved into an in-memory cache that does not provide the ability to scale.
+
+### No cache
+
+Is the cache needed?  Currently most calls for retrieval directly hits the database, meaning that savings could be minimal.  However, this could be an issue if there are a lot calls being made, overwhelming the database.  This would be especially likely if the application has to start scaling due to multiple applications talking to the same database instance.
+
+Additionally, the need for scaling is still not certain, meaning that further thinking on this could just be adding unnecessary complication.  As such, just a simple in-memory cache 
+
+It should be fairly obvious when the database begins to become resource starved and in this case, any form of caching would improve the situation.
+
+#### Distributed cache
+
+This change would require that a call is made to the database to retrieve the Etag value, which would be slower than the current solution when checking the Etag.  A way to mitigate this would be to use an `IDistributedCache` that can be shared between instances, which would avoid the scaling issues as it shares the cache across all instances.
+
+In order to use `IDistributedCache`, a backing cache needs to be selected, which could be Redis or a Postgres cache, though the specifc implementation of this is outside the scope of this RFC.
+
+If the distributed cache wasn't hit, there would be fallback to the database.  The "happy path" for this process would look like this:
+
+```mermaid
+sequenceDiagram
+    actor User 
+    User ->> Api: Retrieve presentation resource
+    Api ->> Cache: Check for Etag
+    alt Not found
+        Cache ->> Database: Retrieve Etag
+        Database ->> Cache: Retrieved Etag
+        Cache -->> Cache: Add to cache
+    end
+    Cache ->> Api: Return Etag
+    Api ->> Api: Additional processing
+    Api ->> User: Return resource with Etag
+```
+
+##### Issues
+
+While using a distributed cache there are some problems that could make using one unviable:
+
+- PUT/POST would benefit, but these operations are far less common and only completed by system administrators, and given the record needs to be retrieved from the database in the current implementation, the savings for this would only be on failed updates, rather than on all calls
+- Increased complexity in management and infrastructure
+- Distributed caches typically live in RAM and so can be expensive to run
+- This is still a call to another external service
+- Not required if the usage of the presentation API is not large
+
+##### Benefits
+
+- Implementation is shared between nodes, meaning it's easy to scale
+- A failed match would not require calls to the database on items in the cache 
+- Ultimately means less calls on the database, which could be a performance bottleneck
+
+#### Publisher and subscriber with internal cache
```

Comment:
fwiw I think, if done correctly and is justified, this is a really powerful pattern. I'm not sure it's warranted here though.